client/map_verifier.go

```diff
@@ -108,11 +110,47 @@ func (m *MapVerifier) VerifyMapLeavesResponse(indexes [][]byte, revision int64,
 	if revision != -1 && int64(mapRoot.Revision) != revision {
 		return nil, status.Errorf(codes.Internal, "got map revision %v, want %v", mapRoot.Revision, revision)
 	}
-	leaves := make([]*trillian.MapLeaf, 0, len(resp.MapLeafInclusion))
-	for _, i := range resp.MapLeafInclusion {
-		if err := m.VerifyMapLeafInclusionHash(mapRoot.RootHash, i); err != nil {
+
+	proofs := make(chan *trillian.MapLeafInclusion)
+	done := make(chan struct{})
+	defer close(done)
+	go func() {
+		defer close(proofs)
+		for _, p := range resp.MapLeafInclusion {
+			select {
+			case proofs <- p:
+			case <-done:
+				return
+			}
+		}
+	}()
+	errors := make(chan error)
+	go func() {
+		defer close(errors)
+		var wg sync.WaitGroup
+		defer wg.Wait()
+		for w := 0; w < runtime.NumCPU(); w++ {
```

Comment:
What are the chances that `len(resp.MapLeafInclusion) < runtime.NumCPU()`? Maybe limit the number of goroutines with the number of items?